src/c2_blockchain/p5_fork_choice.rs

```diff
@@ -196,15 +196,39 @@ fn part_5_heaviest_chain() {
 }
 
 #[test]
-fn part_5_most_even_blocks() {
+fn bc_5_most_even_blocks() {
     let g = Header::genesis();
 
-    let h_a1 = g.child(2, 2);
-    let h_a2 = h_a1.child(2, 2);
+    let mut h_a1 = g.child(2, 0);
+    for i in 0..u64::max_value() {
+        h_a1 = g.child(2, i);
+        if hash(&h_a1) % 2 == 0 {
+            break;
+        }
+    }
+    let mut h_a2 = g.child(2, 0);
+    for i in 0..u64::max_value() {
+        h_a2 = h_a1.child(2, i);
+        if hash(&h_a2) % 2 == 0 {
+            break;
+        }
+    }
```

Comment:
What is the purpose of this change? I'm not fully understanding it. Maybe a comment or two in the test would be helpful?

Reply:
We have to make sure that the hash of the block is even. Otherwise it is random whether this test is correct or not.

Reply:
Very simple way of accomplishing though so if you want me to write it cleaner let me know

Reply:
Ah yeah. I get it now. Before the block hashes may have been even or odd. Good change.